### PR TITLE
fix arrow operator

### PR DESCRIPTION
**Pull request by camio**
Also, added a test illustrating the issue.

Depends on #8.

### CODE REVIEW

**Comment by camio** [comment]
This is a draft until #8 gets merged.

File: include/beman/iterator_interface26/iterator_interface.hpp [hunk 1]
```diff
@@ -131,7 +131,7 @@ template <class Pointer, class Reference, class T>
     requires is_pointer_v<Pointer>
 decltype(auto) make_iterator_pointer(T&& value) // exposition only
 {
-    return addressof(value);
+    return std::addressof(value);
 }
```
Comment (steve-downey) [inline include/beman/iterator_interface26/iterator_interface.hpp:135]: Uninstantiated broken code?

**Review by steve-downey — COMMENTED** [review]
I'll bump up coverage testing on my todo list.

File: include/beman/iterator_interface26/iterator_interface.hpp (hunk 1 above)
Comment (camio) [reply include/beman/iterator_interface26/iterator_interface.hpp:135]: Yep, exactly.

**Review by steve-downey — APPROVED** [review]
+1

**Review by neatudarius — COMMENTED** [review]
LGTM 👍

**Pull request merged**